### ISSUE TITLE
[Feature Request] Require a Worker to specify either workflowsPath or workflowBundle

### ISSUE THREAD

**Issue by ksapchuk** [temporalio/sdk-typescript#542]
### Is your feature request related to a problem? Please describe.
When creating a worker both the workflowPath and workflowBundle are optional. 

https://github.com/temporalio/sdk-typescript/blob/main/packages/worker/src/worker-options.ts#L33-L53
If neither of these are specified the worker happily starts up and does not pick up any work since it's not aware of any workflows it can run. This can be painful to debug since everything appears to be working. 
```
2022-03-16T02:28:38.496Z [INFO] Worker state changed { state: 'RUNNING' }
```

### Describe the solution you'd like
The worker should fail to start without any workflows specified or the types should require one of the options to be passed in.

**Comment by bergundy** [comment]
This is by design, you can run activity only workers.

We could log a warning if that would help.

**Comment by lorensr** [comment]
I was thinking we could also do:

```ts
type WorkerOptionsWithCode = RequireAtLeastOne<WorkerOptions, 'activities' | 'workflowsPath' | 'workflowsBundle' >
```

**Comment by bergundy** [comment]
That could be nice but I fear it would make the API reference for that interface hard to parse.

**Comment by ksapchuk** [comment]
What if the worker could print out what activities or workflows it was able to find? It's not as full proof as type check but at least simplifies debugging.

**Comment by Sushisource** [comment]
@bergundy I opened an item a while back about logging warning. Other users have wasted time on this as well: https://github.com/temporalio/sdk-typescript/issues/471

**Comment by mjameswh** [comment]
Both "no workflows registered" and "no activity registered" results in warning messages being logged (fixed in #808).
